src/analysis/normed_space/real_inner_product.lean

```diff
@@ -0,0 +1,174 @@
+/-
+	Copyright (c) 2019 Zhouhang Zhou. All rights reserved.
+	Released under Apache 2.0 license as described in the file LICENSE.
+	Authors: Zhouhang Zhou
+  -/
+
+import analysis.normed_space.basic
+import analysis.normed_space.discriminant
+
+/-!
+	# Inner Product Space
+
+  Define inner product space over reals and prove its basic properties.
+
+	## Implementation notes
+
+	## Tags
+
+	inner product space, norm
+
+	## References
+  *  [Clément & Martin, *The Lax-Milgram Theorem. A detailed proof to be formalized in Coq*]
+	*  [Clément & Martin, *A Coq formal proof of the Lax–Milgram theorem*]
+
+	The Coq code is available at the following address: http://www.lri.fr/~sboldo/elfic/index.html
+  -/
+
+noncomputable theory
+
+universes u v w
+
+variables {α : Type u} {F : Type v} {G : Type w}
+
+class has_inner (α : Type*) := (inner : α → α → ℝ)
+
+export has_inner (inner)
+
+/--
+An inner product space is a real vector space with an additional operation called inner product.
+Inner product spaces over complex vector space will be defined in another file.
+-/
+class inner_product_space (α : Type*) [add_comm_group α] [vector_space ℝ α] extends has_inner α :=
```

Comment:
I don't understand the difference between this and
```
class inner_product_space (α : Type*) extends add_comm_group α, vector_space ℝ α, has_inner α :=
```

Reply:
Furthermore, @amswerdlow defines inner product space as
```
class herm_inner_product_space (α : Type*) [add_comm_group α] [vector_space ℂ α] extends sesq_form ℂ α conj.ring_invo :=
```
In the real case, it would be 
```
class herm_inner_product_space (α : Type*) [add_comm_group α] [vector_space ℝ α] extends bilinear_form ℝ α  :=
```
I just want to keep it simple so I build inner product space on vector spaces alone, not bilinear form.